src/amo/components/UserProfile/index.js

```diff
@@ -448,7 +448,7 @@ export function mapStateToProps(state: AppState, ownProps: Props) {
   // `getUserByUsername()` requires a string as second argument.
   let user = getUserByUsername(state.users, `${username}`);
 
-  if (!user) {
+  if (!user && /^[0-9]+$/.test(username)) {
```

Comment:
I have been thinking about this patch and one edge case would be `username = 0`.

Reply:
Any username consisting of all numbers would defeat it, yes. I tried it and it's not possible:

<img width="472" alt="screenshot 2018-08-23 12 33 44" src="https://user-images.githubusercontent.com/55398/44541844-d3b40980-a6d0-11e8-90d9-a87755fd9a40.png">

But I tried many numbers at first (including `0`) and they were all taken usernames 😛 That might just be dirty dev data, I'm not sure.

Reply:
Yep but 0 is a special case, no? It will throw an error because of the invariant.

Reply:
Ah, right, probably. We'll see if `0` sends us a bug report. Until then I'm not worried.